AlgoGenesis/C
# Added Maze Solving Algorithm

**My PR NUMBER :** #282 
**Description:**
This pull request introduces a Maze Solving Algorithm implemented in C. The algorithm utilizes depth-first search (DFS) to efficiently find a path from the starting point to the destination in a maze represented as a 2D grid.

**Changes Made:**
Implemented the Maze Solving Algorithm in Maze Solving(Backtracking  folder)
Added helper functions for path exploration and backtracking.
Included comments and documentation for clarity.
Added test cases to validate the functionality of the algorithm.

**How to Test**
Compile the program.c file.
Run the program with a predefined maze layout.
Verify that the algorithm successfully finds a path from the starting point to the destination.

**Please review the code and merge it under gssoc-ext'24 and hactoberfest**

![image](https://github.com/user-attachments/assets/d5ad384e-ae97-4247-ac0e-ddd420e49866)
![image](https://github.com/user-attachments/assets/c707733b-a894-4c32-8cec-7fd9afee6727)
![image](https://github.com/user-attachments/assets/404bfda5-eac3-474d-adbc-a7207494c224)
